### ISSUE TITLE
DefaultValue null is not generated correctly [NULL, "(NULL)", "NULL"]

### ISSUE THREAD

**Issue by md-shah** [sequelize/sequelize-auto#491]
**Versions**
Dialect: SQL Server 16.
Sequelize: 6.3.5
Sequelize-auto: 0.7.8

Given below is my create query for the table. But When I generate the Model for the same, the default values are wrong. Instead of plain JS null, Am getting,

1. NULL: null in uppercase. This will throw error as NULL is undefined.
2. "(NULL)": null is considered as a default function of something
3. "NULL": null as string.

This is my created table query.
```
create table report.report
(
    ID            int identity primary key,
    GUID          varchar(1000) default NULL,
    Name          varchar(1000) default NULL,
    ReportGroupID int           default NULL
)
```

This is the generated Model.
```
const Sequelize = require('sequelize');
module.exports = function (sequelize, DataTypes) {
    return sequelize.define('report', {
        ID: {
            autoIncrement: true,
            type: DataTypes.INTEGER,
            allowNull: false,
            primaryKey: true
        },
        GUID: {
            type: DataTypes.STRING(1000),
            allowNull: true,
            defaultValue: "(NULL)" // null as function call
        },
        Name: {
            type: DataTypes.STRING(1000),
            allowNull: true,
            defaultValue: "(NULL)" // null as function call
        },
        ReportGroupID: {
            type: DataTypes.INTEGER,
            allowNull: true,
            defaultValue: NULL // undefined variable NULL
        }
    });
};

```

In another table, I do have the same issue, but with null as uppercased string => "NULL"

FYI, I do have around 75 tables and only a few of them have this issue (less than 6).

Hope this issue can be identified

**Comment by steveschmitt** [comment]
> Only a few of them have this issue

Any idea what the difference is?  Only a few of them have the `default NULL` explicitly defined?  Or something else?

**Comment by md-shah** [comment]
> >
> Any idea what the difference is? Only a few of them have the `default NULL` explicitly defined? Or something else?


Yes, I did found out the difference. 

The table that had issues are having NULL (Upper case null, BTW which is valid in SQL Server) as a default value, instead of null (In lower case)

This is one of the create table query with no issues.
```
create table company
(
    id_company   int identity (1, 1)
        primary key,
    name         varchar(64)                        not null,
    reference    varchar(64)                        not null
);
```

So is there anything that can be done to make NULL as a special case to make it lowercase or something?

**Comment by steveschmitt** [comment]
Yeah, we should be able to map anything that means null into an actual null.

**Comment by md-shah** [comment]
@steveschmitt  Is there any way that I can contribute? Couldn't find any contribution guidelines. I hope below code will help to solve the issue.

**I think this needs to be added into src/auto-generator.ts (Line: 299)**
```
if (this.dialect.name === "mssql" && (["(NULL)", "NULL"].includes(defaultVal) ||
    typeof defaultVal === "undefined")) {
    defaultVal = null as any; // Override default NULL in MS SQL to javascript null
}
```

**Explanations**

We have 3 cases that might happen. So 2 are handled in the include check and one is checking typeof value
- ` ["(NULL)", "NULL"].includes(defaultVal) ` 

 

> 

> This is to check if the default value is being generated as Stringified NULL. 
> 
>  For example,
> - SQL:  GUID          varchar(1000) default NULL,
> - SQL:  ALTER TABLE [report].[table_name] ADD  DEFAULT (NULL) FOR [Name]


- `typeof defaultVal === "undefined" ` 


> If the null is considerd as NULL, JS will consider it as a variable and we are assigning it into the 'defaultVal variable'. So the value of defaultVal will be undefined.

**Comment by steveschmitt** [comment]
Thanks, I've just added your fix, and it will be in the next release.

**Comment by steveschmitt** [comment]
Fixed in 0.7.10
